README.md

```diff
@@ -20,24 +38,100 @@ How to run
   bin/services
   ```
 
-2. Submit the wordcount-jar
+2. Submit the jar
+  ```bash
+  bin/gear app -jar path/to/streaming-java-template-1.1-SNAPSHOT.jar <app mainclass with package> 
+  ```
+  
+  for example:
+  
   ```bash
-  bin/gear app -jar path/to/streaming-java-template-1.1-SNAPSHOT.jar javatemplate.WordCount
+  bin/gear app -jar target/streaming-java-template-1.1-SNAPSHOT.jar javatemplate.WordCount 
   ```
   
+  
 3. Check the UI
   http://127.0.0.1:8090/  
   
 
-NOTE:
+
+
+> NOTE:
+> 
+> Please use Java7 to run the cluster.
+> 
+> You can set the ENV JAVA_HOME.
+
+> On windows:
+> set JAVA_HOME={path_to_java_7}
+> 
+> On Linux
+> export JAVA_HOME={path_to_java_7}
+
+# Examples description
+
+## kafka2kafka-pipeline
+Very simple example that shows how to read and write from Kafka topics.
+
+The example makes use of Gearpump Connector API, `KafkaSource` and `KafkaSink`, that make simple operations with Kafka super easy.
+
+
+When defining Kafka source, you'll need to provide topic name and zookeeper location:
+```Java
+KafkaSource kafkaSource = new KafkaSource("inputTopic", "localhost:2181");
 ```
-Please use Java7 to run the cluster.
 
-You can set the ENV JAVA_HOME.
+When defining Kafka sink (output), you will just give the destination topic name and Kafka broker address:
+```Java
+KafkaSink kafkaSink = new KafkaSink("outputTopic", "localhost:9092");
+```
 
-On windows:
-set JAVA_HOME={path_to_java_7}
+Keep in mind, that Kafka source processor produces message as byte array (`byte[]`). 
+Also, Kafka sink procesor expects the message to be scala.Tuple. 
 
-on Linux
-export JAVA_HOME={path_to_java_7}
+The example shows dedicated steps that do the necessary conversions. 
+(The conversions don't need to be a separate step, you could include them in other task that do actual computation.)  
+
+### Dependencies
+This example uses zookeeper and Kafka. You need to set them up before running.
+
+Start zookeeper and Kafka:
+
+```bash
+zookeeper/bin/zkServer.sh start
+
+kafka/bin/kafka-server-start.sh kafka/config/server.properties
+```
+
+(Tested with zookeeper 3.4.6 and Kafka 2.11-0.8.2.1. with default settings.)
+
+The app will read messages from `inputTopic` and write to `outputTopic`, so you need to create them beforehand:
```

Comment:
actually, no need to create `outputTopic` beforehand, which is created automatically with default configurations when producing to it.